dask/base.py

```diff
@@ -970,13 +974,15 @@ def normalize_function(func: Callable) -> Callable:
                 if len(function_cache) >= 500:
                     for k in list(function_cache)[::2]:
                         del function_cache[k]
-        function_cache[func] = result
+        function_cache[func] = result  # type: ignore
```

Comment:
if the return type of normalize_function was wrong, why aren't you changing the type of function_cache too?

Reply:
I've unignored this, fixed (simplified) the type annotations, and at the root of the issue I updated the annotation for the `function_cache`. Should be good to go with this one.